### PR TITLE
Multi API generator refactor

### PR DESCRIPTION
**Pull request by mjudeikis**
### Which issue this PR addresses:

Enables to generate multi api swagger more better

### What this PR does / why we need it:

Is cut out version of https://github.com/Azure/ARO-RP/pull/1200
and to fully work needs https://github.com/Azure/ARO-RP/pull/1412

### Test plan for issue:

Np

### Is there any documentation that needs to be updated for this PR?

no

### CODE REVIEW

File: pkg/swagger/types.go [hunk 1]
```diff
@@ -244,6 +244,7 @@ type Schema struct {
 	XML                  *XML                   `json:"xml,omitempty"`
 	ExternalDocs         *ExternalDocumentation `json:"externalDocs,omitempty"`
 	Example              interface{}            `json:"example,omitempty"`
+	SystemData           *Schema                `json:"systemData,omitempty"`
```
Comment (m1kola) [inline pkg/swagger/types.go:247]: This doesn't seem right. Why is system data in the schema? Why is it `SystemData *Schema`?

It should be a part of `OpenShiftCluster` definition. Or am I wrong?

Comment (mjudeikis) [reply pkg/swagger/types.go]: removed :)

**Review by m1kola — DISMISSED** [review]

**Review by m1kola — APPROVED** [review]

**Pull request merged**